src/ray/raylet/node_manager.cc

```diff
@@ -1111,6 +1117,24 @@ void NodeManager::ProcessDisconnectClientMessage(
     }
     // Erase any lease metadata.
     leased_workers_.erase(worker->WorkerId());
+    // Erase leased actor workers and it's indexer
+    auto iter = leased_worker_actor_indexer_.find(worker->WorkerId());
+    if (iter != leased_worker_actor_indexer_.end()) {
+      auto actor_id = iter->second;
+
+      // Report the death event of the worker to gcs server.
+      // Maybe we could use another rpc interface (e.g. NotifyWorkerDeath) to instead of
+      // the `Actors().AsyncUpdate`.
+      auto actor_table_data = std::make_shared<ActorTableData>();
```

Comment:
The `Actors().AsyncUpdate` is a little weird used here, maybe better to use NotifyWorkerDeath to instead.